src/app/berkeley_migration_verifier/berkeley_migration_verifier.ml

```diff
@@ -0,0 +1,516 @@
+end
+
+let test_count = 12
+
+let exit_code = ref 0
+
+module Test = struct
+  type t = { check : Check.t; name : string }
+
+  let of_check check ~name ~idx =
+    { check; name = sprintf "[%d/%d] %s check " idx test_count name }
+
+  let print_errors error_messages =
+    if List.length error_messages > 10 then (
+      Async.printf " Details (truncated to only 10 errors out of %d ): \n"
+        (List.length error_messages) ;
+      List.iter (List.take error_messages 10) ~f:(fun error_message ->
+          Async.printf " - %s\n" error_message ) )
+    else (
+      printf " - Details : \n" ;
+      List.iter error_messages ~f:(fun error_message ->
+          Async.printf "\t%s\n" error_message ) )
+
+  let eval t =
+    match t.check with
+    | Ok ->
+        Async.printf "%s ... ✅ \n" t.name
+    | Error error_messages ->
+        exit_code := !exit_code + 1 ;
+        Async.printf "%s ... ❌ \n" t.name ;
+        print_errors error_messages
+    | Skipped reason ->
+        Async.printf "%s skipping check due to '%s' ... ⚠️ \n" t.name reason
+end
+
+let get_migration_end_slot_for_state_hash ~query_mainnet_db state_hash =
+  let open Deferred.Let_syntax in
+  let%bind maybe_slot =
+    query_mainnet_db ~f:(fun db ->
+        Sql.Mainnet.global_slot_since_genesis_at_state_hash db state_hash )
+  in
+  Deferred.return
+    (Option.value_exn maybe_slot
+       ~message:
+         (Printf.sprintf "Cannot find slot for state hash: %s" state_hash) )
+
+let assert_migrated_db_contains_only_canonical_blocks query_migrated_db height =
+  let%bind canonical_blocks_count =
+    query_migrated_db ~f:(fun db ->
+        Sql.Berkeley.get_all_canonical_blocks_till_height db height )
+  in
+  let%bind all_blocks_till_height =
+    query_migrated_db ~f:(fun db ->
+        Sql.Berkeley.get_all_blocks_till_height db height )
+  in
+  if Int.equal all_blocks_till_height canonical_blocks_count then
+    Deferred.return Check.ok
+  else
+    Deferred.return
+      (Check.err
+         (sprintf "Expected to have only %d canonical block while having %d"
+            canonical_blocks_count all_blocks_till_height ) )
+
+let assert_migrated_db_has_account_accessed query_migrated_db =
+  let%bind account_accessed_count =
+    query_migrated_db ~f:(fun db -> Sql.Berkeley.count_account_accessed db)
+  in
+  if Int.( > ) account_accessed_count 0 then Deferred.return Check.ok
+  else
+    Deferred.return
+      (Check.err
+         (sprintf
+            "Expected to have at least one entry in account accessed table" ) )
+
+let assert_all_user_and_interal_commands_account_ids_appear_in_accounts_accessed
+    query_migrated_db =
+  let%bind account_accessed_in_commands_count =
+    query_migrated_db ~f:(fun db ->
+        Sql.Berkeley.get_account_id_accessed_in_commands db )
+  in
+  let%bind account_accessed =
+    query_migrated_db ~f:(fun db -> Sql.Berkeley.count_account_accessed db)
+  in
+  if Int.equal account_accessed account_accessed_in_commands_count then
```

Comment:
This doesn't quite make sense. This first one counts the number of accounts used gets accessed in the network, while the second is the total number of entries in `accounts_accessed` table. This two can not match.